File: system-test/sec-tests/javascript/sec_06_nosql_injection.js
// SEC-06: NoSQL Injection
// Test: TC-CR-012
app.post('/login', async (req, res) => {
  const { username, password } = req.body;
  const db = client.db('test');

  const user = await db.collection('users').findOne({
    username: username,
    password: { $ne: null }  // $ne bypass
  });

  if (user) res.send('Login success');
  else res.send('Login failed');
});

File: system-test/sec-tests/java/Sec06NoSqlInjection.java
// SEC-06: NoSQL Injection
// Test: TC-CR-012
package com.example;

import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;

public class MongoUserDao {
    public Document findUser(String username, MongoCollection<Document> coll) {
        Document query = new Document("username", username)
            .append("password", new Document("$ne", null));
        return coll.find(query).first();
    }
}
